google/generativeai/client.py

```diff
@@ -37,14 +37,12 @@ def __init__(self, *args, **kwargs):
     def _setup_discovery_api(self):
         api_key = self._client_options.api_key
         if api_key is None:
-            raise ValueError("Uploading to the Files API requires an api key.")
-
-        end_point = self.api_endpoint
+            raise ValueError("Uploading to the File API requires an API key.")
 
         request = googleapiclient.http.HttpRequest(
             http=httplib2.Http(),
             postproc=lambda resp, content: (resp, content),
-            uri=f"https://{end_point}/$discovery/rest?version=v1beta&key={api_key}",
+            uri=f"{GENAI_API_DISCOVERY_URL}?version=v1beta&key={api_key}",
```

Comment:
I needed to lookup the client's end_point here so that would still work in the staging service. 

Couldn't we neec this for testing again in the future?